src/main/scala/mesosphere/marathon/core/pod/MesosContainer.scala

```diff
@@ -2,7 +2,7 @@ package mesosphere.marathon
 package core.pod
 
 import mesosphere.marathon.plugin.ContainerSpec
-import mesosphere.marathon.raml.{Artifact, Endpoint, Image, Lifecycle, MesosExec, Resources}
+import mesosphere.marathon.raml.{Artifact, Endpoint, Image, Lifecycle, LinuxInfo, MesosExec, Resources}
```

Comment:
I know, this is very confusing how we mix raml and state types (I hate this inconsistency) but you **already defined** ` state.LinuxInfo` type (which is correct) and you should use it here and **NOT** the raml type.

Reply:
most of object types with Pods use raml types... the exception as I revisit this is `VolumeMount` .  All other object types are raml... most have a `state` type or area defined in a `state` class for apps.   It seems that raml types were expected for Pods.  I do not understand why Ivan decided to use a `state` defined class... but I would rather stick to the original design unless there is a compelling reason not to.

Reply:
Using the RAML types internally was pure laziness and we should never use them outside of `mesosphere.marathon.raml` and `mesosphere.marathon.api.v2` or decide on one way. I actually removed some references in the past and replaced them with types from `mesosphere.marathon.state`. Furthermore `Artifact`, `Endpoint` etc have no equivalent in `state`. `LinuxInfo` does unless you remove that one.

I know, this is annoying. Ideally one would define the type once and it would be passed all the way through to Mesos. I actually talked with @meichstedt about removing the RAML types (not the spec) and the generator altogether. If we write conversions by hand we might as well write the JSON conversion by hand and not deal with a poorly tested generator. Anyways, this is a midterm thing.

I think there are two options:
* remove `state.LinuxInfo` and use `raml.LinuxInfo` everywhere.
* replace the `raml.LinuxInfo` with `state.LinuxInfo` here

Reply:
What he said ^^